de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend

```diff
@@ -82,4 +88,45 @@ class OvertargetQuickfixProvider extends DefaultQuickfixProvider {
 			}
 		}, 1)
 	}
+	
+	@Inject
+	IResourceSetProvider resourceSetProvider
+	
+	@Fix(OvertargetValidator.FILE_NAME_LIKE_TARGET_NAME)
+	def fixFileNameLikeTargetName(Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, 'Replace with correct tmodel name', '', 'upcase.png') [
+		
+		context |
+			val editor = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor()
+			if (editor instanceof ITextEditor) {
+				val progressMonitor = new NullProgressMonitor()
+				editor.doSave(progressMonitor) //saves the made changes in the file
```

Comment:
Why do we need to save before applying the quickfix?

Reply:
If the editor is not saved the tmodel is not updated and the quickfix uses the name saved in the tmodel. That could lead to errors. Example: tmodelName = test, fileName = newTarget, name in editor Test1: Quickfix: newTarget1. So I think it would be smarter to save the editor to avoid this. Even though it might be a rare case.